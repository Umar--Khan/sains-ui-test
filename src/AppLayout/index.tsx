import { PropsWithChildren, ReactNode } from 'react';

import Header from './Header/Header';
import Main from './Main/Main';

const AppLayout = ({ children }: PropsWithChildren<ReactNode>) => (
  <>
    <Header />
    <Main>{children}</Main>
  </>
);

export default AppLayout;
