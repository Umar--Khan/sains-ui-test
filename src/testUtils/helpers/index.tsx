import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';

import { BasketContext, BasketWrapper } from 'src/context/BasketContext';

export const withRouter = (children: any) => (
  <MemoryRouter>{children}</MemoryRouter>
);

export const renderWithBasketContext = (value: BasketWrapper, children: any) =>
  render(
    <BasketContext.Provider value={value}>
      {withRouter(children)}
    </BasketContext.Provider>,
  );
