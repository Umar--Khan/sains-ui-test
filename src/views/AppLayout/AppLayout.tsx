import { PropsWithChildren, ReactNode } from 'react';

const AppLayout = ({ children }: PropsWithChildren<ReactNode>) => (
  <div>{children}</div>
);

export default AppLayout;
