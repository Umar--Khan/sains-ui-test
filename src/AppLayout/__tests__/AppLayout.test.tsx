import renderer from 'react-test-renderer';
import { withRouter } from 'src/testUtils/helpers';
import AppLayout from '../';

describe('<AppLayout />', () => {
  it('Should render correctly', () => {
    const component = renderer.create(
      withRouter(<AppLayout>App layout test</AppLayout>),
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
