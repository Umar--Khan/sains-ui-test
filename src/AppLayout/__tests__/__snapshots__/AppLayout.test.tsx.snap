// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppLayout /> Should render correctly 1`] = `
Array [
  <header
    className="sc-bdfBwQ JrDbG"
  >
    <div
      className="sc-dlfnbm NLpQA"
    >
      <nav
        className="sc-gsTCUz fHRLXO"
      >
        <a
          aria-current={null}
          className="sc-hKgILt cCXpce"
          href="/store"
          onClick={[Function]}
        >
          Store
        </a>
        <a
          aria-current={null}
          className="sc-hKgILt cCXpce"
          href="/basket"
          onClick={[Function]}
        >
          <span
            data-testid="number-in-basket"
          >
            Basket (0 items)
          </span>
        </a>
      </nav>
    </div>
  </header>,
  <main>
    <div
      className="sc-eCssSg dzUWXM"
    >
      App layout test
    </div>
  </main>,
  <footer
    className="sc-jSgupP iQrmBW"
  >
    <div
      className="sc-gKsewC cXUyCC"
    >
      <span
        className="sc-iBPRYJ jwpiOS"
      >
        Sainsbury’s, Live Well For Less
      </span>
      <div>
        © J Sainsbury plc 
        2021
      </div>
    </div>
  </footer>,
]
`;
